osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -373,6 +350,15 @@ protected override void Update()
                 X = hyperDashTargetPosition;
                 SetHyperDashState();
             }
+
+            if (Dashing || HyperDashing)
+            {
+                double lastTrailTime = trails.LastDashTrail?.LifetimeStart ?? double.NegativeInfinity;
+                double generationInterval = HyperDashing ? 25 : 50;
+
+                if (Time.Current - lastTrailTime >= generationInterval)
+                    trails.DisplayDashTrail(CurrentState, X, Scale * body.Scale, HyperDashing);
```

Comment:
Not sure if pushing down all this data is an improvement. If you cached the catcher and exposed `Scale * body.Scale` on it somehow (as public) the trail display could fetch this stuff all on its own.